### ISSUE TITLE
Tabs aria-selected is not supported on role link

### ISSUE THREAD

**Issue by petermakowski** [canonical-web-and-design/vanilla-framework#4481]
**Describe the bug**

Tabs example suggests using semantics that's not supported and throws an accessibility error.

"aria-selected is not supported on role link"

 We should investigate using different semantics when displaying tabs as links. Maybe we could use `aria-current="section"` instead?

**To Reproduce**

1. Go to https://vanillaframework.io/docs/patterns/tabs#navigation
2. Validate accessibility e.g. with axe devtools
3. See the error thrown

**Additional context**

https://www.stefanjudis.com/blog/aria-selected-and-when-to-use-it/#how-to-not(!)-use-%60aria-selected%60
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current

**Comment by anthonydillon** [comment]
I just ran into this one too while testing MAAS UI. I believe the error is being thrown as `aria-selected` can only be applied to some elements or roles. Follow this example: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected#tab. I believe the only thing missing is `role="tab"` on the selectable element within a container of `role="tablist"`.

**Comment by bartaz** [comment]
Triaged: Possibly we only need to review and update Vanilla examples, so may be a low effort. We should aim to address it in Vanilla v4.
